test/unit/gateways/stripe_test.rb

```diff
@@ -711,6 +711,25 @@ def test_invalid_raw_response
     assert_match(/^Invalid response received from the Stripe API/, response.message)
   end
 
+  def test_api_error_response
+    http_response = stub(code: 500, message: "This is an API error response", body: "")
+    @gateway.expects(:ssl_request).raises(ActiveMerchant::ResponseError.new(http_response))
+
+    assert_raises ActiveMerchant::ResponseError do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end
+  end
+
+  def test_test_mode_live_card_error_response
```

Comment:
This test simply covers `test_mode_live_card` standard error codes are being assigned correctly on `decline_code: test_mode_live_card` Stripe API responses. This is unrelated to this PR and can be moved to a separate PR if it makes more sense.